### PR TITLE
Add batching rule for `pallas_call` (i.e. `vmap`)

### PR DESCRIPTION
**Pull request by sharadmv**

### CODE REVIEW

File: jax_triton/pallas/core.py [hunk 1]
```diff
@@ -52,19 +56,20 @@ def compute_index(self, *args):
...
   index_map_jaxpr: jax_core.ClosedJaxpr
 
   def compute_start_indices(self, loop_idx):
     jaxpr = self.index_map_jaxpr.jaxpr
     consts = self.index_map_jaxpr.consts
     block_indices = jax_core.eval_jaxpr(jaxpr, consts, *loop_idx)
-    return tuple(b * i for b, i in zip(self.block_shape, block_indices))
-
+    return tuple(i if b is mapped else b * i
+                 for b, i in zip(self.block_shape, block_indices))
```
Comment (apaszke) [inline jax_triton/pallas/core.py:67]: Do we have to distinguish between Mapped and a `1` in block_shape?

File: jax_triton/pallas/pallas_call.py [hunk 2]
```diff
@@ -56,14 +59,20 @@
 
 pallas_call_p.def_impl(partial(xla.apply_primitive, pallas_call_p))
 
-def _maybe_dynamic_slice(start_idx, block_shape, value):
+def _maybe_dynamic_slice(start_idx, block_shape, value, is_indexing):
```
Comment (apaszke) [inline jax_triton/pallas/pallas_call.py:62]: It seems like is_indexing can be None? Wouldn't that interact weirdly with the indexing below?

Comment (apaszke) [reply jax_triton/pallas/pallas_call.py:62]: Or does that happen only when start_idx is None? Perhaps add an assertion?

**Review by apaszke — APPROVED** [review]
Can't say I was able to follow this closely, but the parts I understood LGTM

File: jax_triton/pallas/core.py (hunk 1 above)
Comment (sharadmv) [reply jax_triton/pallas/core.py:67]: Yea, mapped means *remove* the dimension whereas `1` keeps it around.

File: jax_triton/pallas/pallas_call.py (hunk 2 above)
Comment (sharadmv) [reply jax_triton/pallas/pallas_call.py:62]: Yea it should only happen when `start_idx` is None. I'll add the assertion.

**Pull request merged**